### PR TITLE
UML- 878 - Account Creation Metric Filter

### PR DESCRIPTION
**Pull request by andrewpearce-digital**
## Purpose
Count account creation events

Fixes UML-878

## Approach

Reorganise log groups so that every environment has it's own
Update application logging so that event_codes are produced on certain events
Create a metric filter resource that increments when matching event_codes appear in logs

## Learning

https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/FilterAndPatternSyntax.html
https://www.terraform.io/docs/providers/aws/r/cloudwatch_log_metric_filter.html

Metric filters increment a count when a matching pattern is found in the logs. I cannot count events that happened before the metric filter was put in place.
Metric filters are applied to log groups. As we consolidate application logs into an account level log group, metric filters are being applied to logs for all dev environments. We can change this pattern and create log groups at the environment level.

## Checklist

* [ ] I have performed a self-review of my own code
* [ ] I have added relevant logging with appropriate levels to my code
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [ ] I have added tests to prove my work
* [ ] The product team have tested these changes

### CODE REVIEW

File: terraform/environment/cloudwatch_metrics.tf [hunk 1]
```diff
@@ -0,0 +1,12 @@
+resource "aws_cloudwatch_log_metric_filter" "account_creation" {
+  name           = "account_creation_messages"
+  pattern        = "{ $.message = \"Account with Id*created using email*\" }"
```
Comment (cooperaj) [inline terraform/environment/cloudwatch_metrics.tf:3]: Would it be useful to have a particular code associated with this? i.e. ACCOUNT_CREATED

Comment (cooperaj) [reply terraform/environment/cloudwatch_metrics.tf:3]: I think we already use code for certain errors. we can move to event if that makes more sense.

Comment (cooperaj) [reply terraform/environment/cloudwatch_metrics.tf:3]: Mind if I work on this PR to add those?

**Comment by andrewpearce-digital** [comment]
image

**Comment by cooperaj** [comment]
https://stackoverflow.com/questions/41051769/how-can-i-access-aws-cloudwatch-custom-metrics-via-php-sdk

File: terraform/environment/cloudwatch_metrics.tf (hunk 1 above)
Comment (andrewpearce-digital) [reply terraform/environment/cloudwatch_metrics.tf:3]: Didn't see this message due to needing a refresh, You're welcome to pull and have a go.

File: terraform/environment/terraform.tfvars.json [hunk 2]
```diff
@@ -17,7 +17,8 @@
       "cookie_expires_use" : 1440,
       "logging_level": 100,
       "pagerduty_service_name": "Use a Lasting Power of Attorney Non-Production",
-      "use_legacy_codes_service": false
+      "use_legacy_codes_service": true,
```
Comment (williamfalconeruk) [inline terraform/environment/terraform.tfvars.json:20]: was this meant to be turned back to legacy mode in dev?

Comment (andrewpearce-digital) [reply terraform/environment/terraform.tfvars.json:20]:
```suggestion
      "use_legacy_codes_service": false,
```

**Pull request merged**